src/test/java/uk/gov/ons/census/casesvc/messaging/QuestionnaireLinkedReceiverIT.java

```diff
@@ -586,9 +587,13 @@ private void validateEvents(
     assertThat(event.getEventDescription()).isEqualTo("Questionnaire Linked");
 
     JSONObject payload = new JSONObject(event.getEventPayload());
-    assertThat(payload.length()).isEqualTo(2);
+    assertThat(payload.length()).isEqualTo(payloadLength);
     assertThat(payload.getString("caseId")).isEqualTo(TEST_CASE_ID.toString());
     assertThat(payload.getString("questionnaireId")).isEqualTo(expectedQuestionnaireId);
+    if (payloadLength == 3) {
```

Comment:
Not sure about using the payload length like this, seems like the sort of thing that will make the tests quite brittle.

Reply:
Spoke with Luke and made some little changes to hopefully make this better